.circleci/config.yml

```diff
@@ -9,31 +9,35 @@ jobs:
     executor: docker-executor
     environment:
       JVM_OPTS: -Xmx512m
-      TERM: dumb
+      TERM: dumb  
+      GITHUB_TOKEN: ${GITHUB_TOKEN}  # Add the GitHub token as an environment variable
 
     steps:
       - checkout
-      # Install Docker to build and push the image
       - setup_remote_docker:
           version: 20.10.14
-
-      # Build the Docker image
       - run:
-          name: Build Docker image
+          name: Build and Push Docker tag Image
           command: |
-            ./gradlew bootJar
-            docker build -t openmf/ph-ee-connector-ams-mifos:${CIRCLE_TAG} .
+            # Set environment variables
+            IMAGE_TAG=$CIRCLE_TAG
 
-      # Log in to DockerHub using environment variables
-      - run:
-          name: Login to DockerHub
-          command: echo "${DOCKERHUB_PASSWORD}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+            # Check if the Docker image with the same tag already exists in Docker Hub
+            if curl -s -f -u "$DOCKERHUB_USERNAME":"$DOCKERHUB_PASSWORD" "https://hub.docker.com/v2/repositories/openmf/ph-ee-connector-ams-mifos/tags/$IMAGE_TAG" > /dev/null; then
```

Comment:
If you externalize the project name as a project specific env variable, it will be easy to maintain this code